File: go/delete-greatest-value-in-each-row.go
func deleteGreatestValue(grid [][]int) int {
	output := 0
	for _, row := range grid {
		sort.Ints(row)
	}
	for i := 0; i < len(grid[0]); i++ {
		maximal := 0
		for _, row := range grid {
			if maximal < row[i] {
				maximal = row[i]
			}
		}
		output += maximal
	}
	return output
}

File: python/delete-greatest-value-in-each-row.py
class Solution:
    def deleteGreatestValue(self, grid: List[List[int]]) -> int:
        output = 0
        for row in grid:
            row.sort()
        while 0 < len(grid[0]):
            maximal = 0
            for row in grid:
                maximal = max(maximal, row.pop())
            output += maximal
        return output
